workflow/controller/controller.go

```diff
@@ -507,19 +507,50 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 				return err
 			}
 		case labelPodCompleted:
-			_, err := pods.Patch(
+			data, err := json.Marshal(map[string]interface{}{
+				"metadata": map[string]interface{}{
+					"labels": map[string]interface{}{
+						common.LabelKeyCompleted: "true",
+					},
+					"finalizers": []string{},
```

Comment:
I think other processes can add finalizer. We should only remove ours.

Reply:
@alexec done